scene/3d/collision_object.h

```diff
@@ -38,7 +38,17 @@ class CollisionObject : public Spatial {
 
 	GDCLASS(CollisionObject, Spatial);
 
-	bool area;
+public:
+	enum CollisionObjectType {
```

Comment:
I am still not convinced on this, given SoftBody does not use shapes. 

Also not convinced on modifying MeshInstance for this.

Does it not make more sense that SoftBody does everything as a single node and inherits GeometryInstance? This way there is no need to modify neither of these nodes to add softbody.

Reply:
The thing is that softbody use as collision shape the mesh that is rendered, but I agree with you this approach so good. Also inherit from GeometryInstance is not a good thing since softbody is a physical object mainly. A solution to this could be set the mesh resource directly to the soft body instead to use a MeshInstance, and let do all work inside the SoftBody

Reply:
But on the other hand the other problems are that I need to copy all functions from MeshInstance in order to allow artist to set Material and other things related to rendering..